perf/impl/go-libp2p/v0.29/perf.go

```diff
@@ -100,15 +121,49 @@ func sendBytes(s io.Writer, bytesToSend uint64) error {
 			return err
 		}
 		bytesToSend -= uint64(n)
+		lastReportWrite += n
 	}
 	return nil
 }
 
 func drainStream(s io.Reader) (uint64, error) {
 	var recvd int64
-	recvd, err := io.Copy(io.Discard, s)
+	recvd, err := io.Copy(io.Discard, & reportingReader { orig: s, LastReportTime: time.Now() })
```

Comment:
Does it even work with the spaces? Can you run `gofmt ./...` on the code?
```suggestion
	recvd, err := io.Copy(io.Discard, &reportingReader{ orig: s, LastReportTime: time.Now() })
```